dask/task_spec.py

```diff
@@ -0,0 +1,742 @@
+from __future__ import annotations
+
+""" Task specification for dask
+
+This module contains the task specification for dask. It is used to represent
+runnable and non-runnable (i.e. literals) tasks in a dask graph.
+
+Simple examples of how to express tasks in dask
+-----------------------------------------------
+
+.. code-block:: python
+
+    func("a", "b") ~ Task("key", func, ["a", "b"])
+
+    [func("a"), func("b")] ~ SequenceOfTasks(
+        "key",
+        [Task("key-1", func, "a"), Task("key-2", func, "b")]
+    )
+
+    {"a": func("b")} ~ DictOfTasks("key", {"a": Task("a", func, "b")})
+
+    "literal-string" ~ Literal("key", "literal-string")
+
+
+Keys, Aliases and KeyRefs
```

Comment:
How would I rename a result that was previously stored under a different key?

``{"key": KeyRef("old-key")}``?

Reply:
yes, this